lib/statsd/instrument/strict.rb

```diff
@@ -0,0 +1,103 @@
+    UNSPECIFIED = Object.new.freeze
+    private_constant :UNSPECIFIED
+
+    # The Strict monkeypatch can be loaded to see if you're using the StatsD library in
+    # a deprecated way.
+    #
+    # - The metric methods are not retuning a Metric instance.
+    # - Only accept keyword arguments for tags and sample_rate, rather than position arguments.
+    # - Only accept a position argument for value, rather than a keyword argument.
+    # - The provided arguments have the right type.
+    #
+    # You can enable thois monkeypatch by changing your Gemfile as follows:
+    #
+    #     gem 'statsd-instrument', require: 'statsd/instrument/strict'
+    #
+    # By doing this as part of your QA/CI, you can find where you are still using deprecated patterns,
+    # and fix them before the deprecated behavior is removed in the next major version.
+    #
+    # This monkeypatch is not meant to be used in production.
+    module Strict
+      def increment(key, value = 1, sample_rate: nil, tags: nil, prefix: StatsD.prefix, no_prefix: false)
+        raise ArgumentError, "StatsD.increment does not accept a block" if block_given?
+        raise ArgumentError, "The value argument should be an integer, got #{value.inspect}" unless value.is_a?(Numeric)
+        check_tags_and_sample_rate(sample_rate, tags)
+
+        super
+      end
+
+      def gauge(key, value, sample_rate: nil, tags: nil, prefix: StatsD.prefix, no_prefix: false)
+        raise ArgumentError, "StatsD.increment does not accept a block" if block_given?
+        raise ArgumentError, "The value argument should be an integer, got #{value.inspect}" unless value.is_a?(Numeric)
+        check_tags_and_sample_rate(sample_rate, tags)
+
+        super
+      end
+
+      def histogram(key, value, sample_rate: nil, tags: nil, prefix: StatsD.prefix, no_prefix: false)
+        raise ArgumentError, "StatsD.increment does not accept a block" if block_given?
+        raise ArgumentError, "The value argument should be an integer, got #{value.inspect}" unless value.is_a?(Numeric)
+        check_tags_and_sample_rate(sample_rate, tags)
+
+        super
+      end
+
+      def set(key, value, sample_rate: nil, tags: nil, prefix: StatsD.prefix, no_prefix: false)
+        raise ArgumentError, "StatsD.set does not accept a block" if block_given?
+        check_tags_and_sample_rate(sample_rate, tags)
+
+        super
+      end
+
+      def measure(key, value = UNSPECIFIED, sample_rate: nil, tags: nil,
+        prefix: StatsD.prefix, no_prefix: false, as_dist: false, &block)
+
+        check_block_or_numeric_value(value, &block)
+        check_tags_and_sample_rate(sample_rate, tags)
+
+        super
+      end
+
+      def distribution(key, value = UNSPECIFIED, sample_rate: nil, tags: nil,
+        prefix: StatsD.prefix, no_prefix: false, &block)
+
+        check_block_or_numeric_value(value, &block)
+        check_tags_and_sample_rate(sample_rate, tags)
+
+        super
+      end
+
+      protected
```

Comment:
Looks like this also could be `private`

```suggestion
      private
```